torch/testing/_internal/distributed/_tensor/common_dtensor.py

```diff
@@ -31,17 +35,27 @@
 )
 
 from torch.utils._pytree import tree_flatten, tree_unflatten, TreeSpec
+from torch._utils import _get_device_module
+
+DEVICE_COUNT = _get_device_module("cuda").device_count() if TEST_CUDA else _get_device_module("hpu").device_count()
 
 DEVICE_TYPE = (
-    "cuda" if torch.cuda.is_available() and torch.cuda.device_count() > 1 else "cpu"
+    "cuda"
+    if TEST_CUDA and DEVICE_COUNT > 1
+    else "hpu"
+    if TEST_HPU and DEVICE_COUNT > 1
+    else "cpu"
+)
```

Comment:
I wonder if we could first define `DEVICE_TYPE` then `DEVICE_COUNT` so that we can call `_get_device_module(DEVICE_TYPE)` then `device_count()`?